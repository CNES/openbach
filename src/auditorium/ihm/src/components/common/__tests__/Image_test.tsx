import * as React from "react";
import * as ReactDOM from "react-dom";
import Image from "../Image";

import * as TestUtils from "react-addons-test-utils";

import * as renderer from "react-test-renderer";

test("Image check snapshot", () => {
    const component = renderer.create(
        <Image src="http://www.facebook.com" />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("Link changes the class when hovered", () => {
    const component = TestUtils.renderIntoDocument(
        <Image src="http://www.facebook.com" />,
    );

    const componentNode = ReactDOM.findDOMNode(component);
    expect(componentNode.textContent).toEqual("");
});
