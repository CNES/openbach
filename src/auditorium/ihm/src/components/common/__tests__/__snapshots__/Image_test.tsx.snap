exports[`test Image check snapshot 1`] = `
<div
  style={
    Object {
      "backgroundImage": "url(\"http://www.facebook.com\")",
      "backgroundPosition": "center center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "contain",
      "height": 100,
      "width": 100,
    }
  } />
`;
